plugins/modules/accesspoint_workflow_manager.py

```diff
@@ -2816,16 +3409,22 @@ def update_ap_configuration(self, ap_config):
             ap_config["adminStatus"] = True \
                 if ap_config["adminStatus"] == "Enabled" else False
 
-        if ap_config.get(self.keymap["ap_name"]) is not None:
-            temp_dict[self.keymap["ap_name"]] = ap_config.get(self.keymap["ap_name"])
-            temp_dict["apNameNew"] = ap_config["apNameNew"]
-            temp_dict[self.keymap["mac_address"]] = ap_config[self.keymap["mac_address"]]
-            del ap_config[self.keymap["ap_name"]]
-            del ap_config["apNameNew"]
-        elif ap_config.get(self.keymap["mac_address"]) is not None:
-            temp_dict[self.keymap["mac_address"]] = ap_config.get(self.keymap["mac_address"])
-
-        ap_config["apList"].append(temp_dict)
+        if not ap_config.get("bulk_update"):
+            if ap_config.get(self.keymap["ap_name"]) is not None:
+                temp_dict[self.keymap["ap_name"]] = ap_config.get(self.keymap["ap_name"])
+                temp_dict["apNameNew"] = ap_config["apNameNew"]
+                temp_dict[self.keymap["mac_address"]] = ap_config[self.keymap["mac_address"]]
+                del ap_config[self.keymap["ap_name"]]
+                del ap_config["apNameNew"]
+            elif ap_config.get(self.keymap["mac_address"]) is not None:
+                temp_dict[self.keymap["mac_address"]] = ap_config.get(self.keymap["mac_address"])
+            ap_config["apList"].append(temp_dict)
```

Comment:
Add else part.. if both are none.. We know it should not reach this point.. But adding a log here may help later..

    else:
        self.log("No AP name or MAC address found in ap_config.", "WARNING")

    ap_config["apList"].append(temp_dict)
    self.log("Added temp_dict to apList. Current apList: {}".format(self.pprint(ap_config["apList"])), "INFO")